(DOCS-3029) TLS deprecation doc (#12888)
DataDog/documentation: content/en/agent/guide/tls_deprecation_1_2.md

---
title: Deprecation notice for TLS version < 1.2
kind: guide
---

## Overview

> Transport Layer Security (TLS) is a critical security protocol used to protect web traffic. It provides confidentiality and integrity of data in transit between clients and servers exchanging information. Datadog is disabling support for older versions of TLS, below 1.2 (SSLv3, TLS v1.0, TLS v1.2) across public facing Datadog applications, beginning March 31st. If you are using unsupported clients to connect to Datadog after older protocols are disabled, you will begin receiving connection error messages.

Author: ```suggestion
Transport Layer Security (TLS) is a critical security protocol used to protect web traffic. It provides confidentiality and integrity of data in transit between clients and servers exchanging information. Datadog is disabling support for older versions of TLS, below 1.2 (SSLv3, TLS v1.0, TLS v1.2) across public facing Datadog applications, beginning March 31, 2022. If you are using unsupported clients to connect to Datadog after older protocols are disabled, you will begin receiving connection error messages.
```

> Transport Layer Security (TLS) is a critical security protocol used to protect web traffic. It provides confidentiality and integrity of data in transit between clients and servers exchanging information. Datadog is disabling support for older versions of TLS, below 1.2 (SSLv3, TLS v1.0, TLS v1.2) across public facing Datadog applications, beginning March 31, 2022. If you are using unsupported clients to connect to Datadog after older protocols are disabled, you will begin receiving connection error messages.

Reviewer: ```suggestion
Transport Layer Security (TLS) is a critical security protocol used to protect web traffic. It provides confidentiality and integrity of data in transit between clients and servers exchanging information. Datadog is disabling support for older versions of TLS, below 1.2 (SSLv3, TLS v1.0, TLS v1.2) across public facing Datadog applications, beginning March 31, 2022. If you use unsupported clients to connect to Datadog after the older protocols are disabled, you will receive connection error messages.
```

### Reason for deprecation

These protocols are being deprecated to ensure that customers connect to Datadog using secure connection channels. This is in accordance with a decision from the Internet Engineering Task Force (IETF) to deprecate these protocols as of March 25, 2021. ([https://datatracker.ietf.org/doc/rfc8996/][1])

## Browser support

> Modern browsers have had support for TLS 1.2 for a while, view the "Can I use..." [compatibility matrix][2] to determine if your specific browser and version are affected.

Reviewer: ```suggestion
Modern browsers have had support for TLS 1.2 for a while. See the "Can I use..." [compatibility matrix][2] to determine if your specific browser and version are affected.
```

## Language support

### Python

> Starting in Python 3.10, the SSL module defaults to TLS 1.2 and the use of modern cipher suites, see [][3].

Reviewer: missed link here

> Starting in Python 3.10, the SSL module defaults to TLS 1.2 and the use of modern cipher suites, see [][3].

Author: ```suggestion
Starting in Python 3.10, the SSL module defaults to TLS 1.2 and the use of modern cipher suites, see [https://bugs.python.org/issue43998][3].
```

### Golang

If you are using a newer version of Go (1.13 or above), Go already supports TLS 1.2 by default, so no changes are necessary.

When using older versions of Go, set your TLS Client Configuration’s MinVersion to explicitly use TLS 1.2:

```
TLSClientConfig: &tls.Config{
        MinVersion: tls.VersionTLS12,
    }
```

### Java

If your application runs on Java 1.7 or Java 1.6 (update 111 or later), you can set the `https.protocols` system property when starting the JVM to enable additional protocols for connections made using the `HttpsURLConnection` class. For example, by setting
`Dhttps.protocols=TLSv1.2`.

> If your application runs on Java 1.6 prior to update 111, or earlier, TLS 1.1 and 1.2 are not supported. Therefore, you need to update the version of Java your application runs on. If you use another library for connections such as Apache HttpClient, see the Apache-HttpClient section below.

Author: Original google docs mentions "see Apache-HttpClient section below" but there's no section below. After some quick googling, I couldn't find any official looking docs from apache on this topic, and it probably varies by library.

> If your application runs on Java 1.6 prior to update 111, or earlier, TLS 1.1 and 1.2 are not supported. Therefore, you need to update the version of Java your application runs on. If you use another library for connections such as Apache HttpClient, see the Apache-HttpClient section below.

Author: ```suggestion
If your application runs on Java 1.6 prior to update 111, or earlier, TLS 1.1 and 1.2 are not supported. Therefore, you need to update the version of Java your application runs on.
```
